libkbfs/key_bundle_v3.go

```diff
@@ -50,83 +51,62 @@ func (dkimV3 DeviceKeyInfoMapV3) fillInDeviceInfos(crypto cryptoPure,
 	return serverHalves, nil
 }
 
-func (dkimV3 DeviceKeyInfoMapV3) toDKIM(codec kbfscodec.Codec) (
-	DeviceKeyInfoMap, error) {
-	dkim := make(DeviceKeyInfoMap, len(dkimV3))
-	for key, info := range dkimV3 {
-		var infoCopy TLFCryptKeyInfo
-		err := kbfscodec.Update(codec, &infoCopy, info)
-		if err != nil {
-			return nil, err
-		}
-		dkim[key] = infoCopy
+func (dkimV3 DeviceKeyInfoMapV3) toPublicKeys() DevicePublicKeys {
+	publicKeys := make(DevicePublicKeys, len(dkimV3))
+	for key := range dkimV3 {
+		publicKeys[key] = true
 	}
-	return dkim, nil
-}
-
-func dkimToV3(codec kbfscodec.Codec, dkim DeviceKeyInfoMap) (
-	DeviceKeyInfoMapV3, error) {
-	dkimV3 := make(DeviceKeyInfoMapV3, len(dkim))
-	for key, info := range dkim {
-		var infoCopy TLFCryptKeyInfo
-		err := kbfscodec.Update(codec, &infoCopy, info)
-		if err != nil {
-			return nil, err
-		}
-		dkimV3[key] = infoCopy
-	}
-	return dkimV3, nil
+	return publicKeys
 }
 
 // UserDeviceKeyInfoMapV3 maps a user's keybase UID to their
 // DeviceKeyInfoMapV3.
 type UserDeviceKeyInfoMapV3 map[keybase1.UID]DeviceKeyInfoMapV3
 
-func (udkimV3 UserDeviceKeyInfoMapV3) toUDKIM(
-	codec kbfscodec.Codec) (UserDeviceKeyInfoMap, error) {
-	udkim := make(UserDeviceKeyInfoMap, len(udkimV3))
+func (udkimV3 UserDeviceKeyInfoMapV3) toPublicKeys() UserDevicePublicKeys {
+	publicKeys := make(UserDevicePublicKeys, len(udkimV3))
 	for u, dkimV3 := range udkimV3 {
-		dkim, err := dkimV3.toDKIM(codec)
-		if err != nil {
-			return nil, err
-		}
-		udkim[u] = dkim
+		publicKeys[u] = dkimV3.toPublicKeys()
 	}
-	return udkim, nil
+	return publicKeys
 }
 
-func udkimToV3(codec kbfscodec.Codec, udkim UserDeviceKeyInfoMap) (
+func writerUDKIMV2ToV3(codec kbfscodec.Codec, udkimV2 UserDeviceKeyInfoMapV2,
+	ePubKeyCount int) (
 	UserDeviceKeyInfoMapV3, error) {
-	udkimV3 := make(UserDeviceKeyInfoMapV3, len(udkim))
-	for u, dkim := range udkim {
-		dkimV3, err := dkimToV3(codec, dkim)
-		if err != nil {
-			return nil, err
+	udkimV3 := make(UserDeviceKeyInfoMapV3, len(udkimV2))
+	for uid, dkimV2 := range udkimV2 {
+		dkimV3 := make(DeviceKeyInfoMapV3, len(dkimV2))
+		for kid, info := range dkimV2 {
+			index := info.EPubKeyIndex
+			if index < 0 || index >= ePubKeyCount {
+				return nil, fmt.Errorf(
```

Comment:
Right, I wasn't sure if that was allowed -- discussed more in https://keybase.atlassian.net/browse/KBFS-1719 .